tests/test_mongodb_table_reset_log.py

```diff
@@ -0,0 +1,209 @@
+                                                                   self.expected_sync_streams(),
+                                                                   self.expected_pks())
+
+        # Verify that the full table was synced
+        for tap_stream_id in self.expected_sync_streams():
+            self.assertGreaterEqual(record_count_by_stream[tap_stream_id],self.expected_row_counts()[tap_stream_id])
+
+        # Verify that we have 'initial_full_table_complete' bookmark
+        state = menagerie.get_state(conn_id)
+        first_versions = {}
+
+        for tap_stream_id in self.expected_check_streams():
+            # assert that the state has an initial_full_table_complete == True
+            self.assertTrue(state['bookmarks'][tap_stream_id]['initial_full_table_complete'])
+            # assert that there is a version bookmark in state
+            first_versions[tap_stream_id] = state['bookmarks'][tap_stream_id]['version']
+            self.assertIsNotNone(first_versions[tap_stream_id])
+            # Verify that we have a oplog_ts_time and oplog_ts_inc bookmark
+            self.assertIsNotNone(state['bookmarks'][tap_stream_id]['oplog_ts_time'])
+            self.assertIsNotNone(state['bookmarks'][tap_stream_id]['oplog_ts_inc'])
+
+
+        # manipulate state to simulate table reset
+        reset_stream = 'simple_db-simple_coll_2'
+        state['bookmarks'].pop(reset_stream)
+        menagerie.set_state(conn_id, state)
+
+
+        #  -------------------------------------------
+        #  ----------- Subsequent Oplog Sync ---------
+        #  -------------------------------------------
+
+        # Run sync
+        sync_job_name = runner.run_sync_mode(self, conn_id)
+
+        exit_status = menagerie.get_exit_status(conn_id, sync_job_name)
+        menagerie.verify_sync_exit_status(self, exit_status, sync_job_name)
+
+        # verify the persisted schema was correct
+        messages_by_stream = runner.get_records_from_target_output()
+        records_by_stream = {}
+        for stream_name in self.expected_sync_streams():
+            records_by_stream[stream_name] = [x for x in messages_by_stream[stream_name]['messages'] if x.get('action') == 'upsert']
+
+        # assert that each of the streams that we synced are the ones that we expect to see
+        record_count_by_stream = runner.examine_target_output_file(self,
+                                                                   conn_id,
+                                                                   self.expected_sync_streams(),
+                                                                   self.expected_pks())
+
+        # Verify the expected number of records per table
+        for k,v in record_count_by_stream.items():
+            if k == 'simple_coll_1':
+                self.assertEqual(v, 0) # not reset
+            if k == 'simple_coll_2':
+                self.assertEqual(v, 100) # reset stream
```

Comment:
Same as other test, could you add an assertion to verify the state is set correctly following the sync of the reset stream?

Reply:
Ok